tests/main/hook-permissions/task.yaml

```diff
@@ -0,0 +1,44 @@
+summary: Ensure that post-refresh hook has permissions for its connected slot.
+
+# Note: it is critical for this test to use a slot that requires
+# peer=<plug security tag> expression label for dbus, and not simply
+# label=unconfined on a permanent slot snippet.
+details: |
+    Ensure that post-refresh hook has permissions to query upower-observe slot
+    and enumerate upower devices.
+
+systems: [-fedora-*, -opensuse-*, -arch-*, -amazon-*, -centos-*]
+
+restore: |
+    rm -f ./enumerate.txt
+    snap remove --purge test-snap || true
+    # shellcheck source=tests/lib/systems.sh
+    . "$TESTSLIB/systems.sh"
+    if is_core_system; then
```

Comment:
should not be necesary to remove snaps on restore because we do it on reset.sh automatically. If you are doing if for a specific reason please leave a comment so it is not deleted in the future.